src/sentry/utils/jwt.py

```diff
@@ -0,0 +1,84 @@
+"""Common handling of JWT tokens.
+
+This is an attempt to have all the interactions with JWT in once place, so that we have once
+central place which handles JWT in a uniform way.
+"""
+
+from typing import List, Mapping, Optional, Union
+
+import jwt as pyjwt
+from cryptography.hazmat.primitives.asymmetric.rsa import RSAPrivateKey, RSAPublicKey
+from jwt import DecodeError
+
+__all__ = ["peek_claims", "decode", "encode", "authorization_header", "DecodeError"]
+
+
+def peek_claims(token: str) -> Mapping[str, str]:
+    """Returns the claims (payload) in the JWT token without validation.
+
+    These claims can be used to look up the correct key to use in :func:`decode`.
+    """
+    return pyjwt.decode(token, verify=False)
+
+
+def decode(
+    token: str,
+    key: bytes,
+    *,
+    audience: Union[None, str, bool] = None,
```

Comment:
This type is now perhaps a bit weird, but for a rust developer this makes total sense.  You can leave it unset (meaning validate so if it is present in the claims it will fail validation), tell it to not validate or provide the value to validate with.  All modes are used in the code and I think this is more readable than combining this into two options.

Reply:
Yes this is weird but I think it's OK as long as it's explained in the docstring.
Can you add `token`, `key`, and `algorithms` to the docstring?
Also it'd be clearer if the type of `audience` was:
```
audience: Optional[Union[str, bool]] = None,
```